media/public/include/IMediaPipeline.h

```diff
@@ -127,10 +140,36 @@ class IMediaPipeline
 
         /**
          * @brief Return the MIME type.
+         */
+        std::string getMimeType() const { return m_mimeType; }
+
+        /**
+         * @brief Gets the segment alignment
+         */
+        SegmentAlignment getSegmentAlignment() const { return m_alignment; }
+
+        /**
+         * @brief Sets the segment alignment
          *
-         * @warning API is likely to change to be MIME-type based
+         * @param[in] alignment : The new segment alignment
          */
-        std::string getCaps() const { return m_caps; }
+        void setSegmentAlignment(const SegmentAlignment &alignment) { m_alignment = alignment; }

```

Comment:
I know this is how I specified the API but it implies that the client can change the alignment on this fly. I wonder if it would be better to only allow setting this in the constructor? Or have the API return a bool so we can disallow setting it more that once/after streaming has started. @aczs - what do you think?